ci/official/any.sh

```diff
@@ -0,0 +1,33 @@
+#!/bin/bash
+source "${BASH_SOURCE%/*}/utilities/setup.sh"
+
+# Parse options and build targets into arrays, so that shelllint doesn't yell
+# about readability. We can't pipe into 'read -ra' to create an array because
+# piped commands run in subshells, which can't store variables outside of the
+# subshell environment.
+# Ignore grep failures since we're using it for basic filtering
+set +e
+filtered_build_targets=( $(echo "$BUILD_TARGETS" | tr ' ' '\n' | grep . | tee build_targets.txt) )
+nonpip_targets=( $(echo "$TEST_TARGETS" | tr ' ' '\n' | grep -E "^//tensorflow/" | tee nonpip_targets.txt) )
+config=( $(echo "$CONFIG_OPTIONS" ) )
+test_flags=( $(echo "$TEST_FLAGS" ) )
+set -e
+
+[[ "$TFCI_NVIDIA_SMI_ENABLE" == 1 ]] && tfrun nvidia-smi
+
+if [[ -s build_targets.txt ]]; then
+  tfrun bazel "${TFCI_BAZEL_BAZELRC_ARGS[@]}" "${config[@]}" "${filtered_build_targets[@]}"
+fi
+
+if [[ "${PIP_WHEEL}" -eq "1" ]]; then
+  # Update the version numbers to build a "nightly" package
+  [[ "$TFCI_NIGHTLY_UPDATE_VERSION_ENABLE" == 1 ]] && tfrun python3 tensorflow/tools/ci_build/update_version.py --nightly
+
+  tfrun bazel "${TFCI_BAZEL_BAZELRC_ARGS[@]}" build "${TFCI_BAZEL_CACHE_ARGS[@]}" tensorflow/tools/pip_package:build_pip_package
```

Comment:
Should `TFCI_BAZEL_CACHE_ARGS` be even more general?  It seems to fit any set of additional configs that want to be handed to all bazel invocations.

Reply:
Hmm, this seems pretty tricky. I agree that it is possible to put a lot more general stuff in here, but I am wary of constructing the variables such that it's possible to put way too much behavior in one configuration option. For example, it's possible that a TFCI_BAZEL_COMMON_CONFIG_ARGS could, through a lazy "I need to add this in a hurry" CL, become...

```
( --config rbe --flaky_test_attempts 2 --config build_event_export --config ...)
```

...because all of those could also be applied to every Bazel invocation. It could become overloaded and confusing, if we dump too much stuff into there. So I don't want that to happen, but on the other hand I also don't want...

```
TFCI_BAZEL_CONFIG_CACHE_ARGS=
TFCI_BAZEL_CONFIG_RBE_ENABLE=
TFCI_BAZEL_CONFIG_BUILD_EVENT_EXPORT_ENABLE=
TFCI_BAZEL_CONFIG_FLAKY_TEST_ATTEMPTS=
TFCI_BAZEL_CONFIG_EVERY_POSSIBLE_OPTION_FOR_EVERYTHING=
```
...which is just as feasible and actually much worse. So I think I'll go with TF_BAZEL_COMMON_CONFIG_ARGS and just document it carefully.

Reply:
(Currently not documented, but will be in the future. I haven't figured out where to document all the env variables yet -- probably in a "sample.env" or something)